app/controllers/hmis_api/projects_controller.rb

```diff
@@ -0,0 +1,45 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# TEMPORARY: example of a POST request for testing HMIS authentication. Replace with GraphQL endpoint!
+
+class HmisApi::ProjectsController < HmisApi::BaseController
+  include ArelHelper
+
+  def index
+    render json: project_scope.pluck(:ProjectName)
+  end
+
+  private def project_types
+    return HUD.project_types.keys unless project_params[:project_types].present?
+
+    @project_types ||= begin
+      types = []
+
+      project_type_to_id = project_source::PERFORMANCE_REPORTING.merge(project_source::RESIDENTIAL_PROJECT_TYPES)
+      if project_params[:project_types].present?
+        project_params[:project_types]&.select(&:present?)&.map(&:to_sym)&.each do |type|
+          types += project_type_to_id[type]
+        end
+      end
+      types
+    end
+  end
+
+  def project_params
+    params.permit(
+      project_types: [],
+    )
+  end
+
+  def project_source
+    GrdaWarehouse::Hud::Project
+  end
+
+  private def project_scope
+    project_source.viewable_by(current_hmis_api_user).with_project_type(project_types).distinct
```

Comment:
When you login with the API, it doesn't setup `current_user` but sets up `current_hmis_api_user`.  Did I get that right?  Any reason not to re-use `current_user`?

Reply:
Yeah that's correct. I set up class `HmisApiUser` which is a subclass of `User` so we can add any different functionality we want there. Right now the only difference is the `skip_session_limitable` behavior, but I can imagine we may want other differences in behavior later on. We really shouldn't have any endpoint crossover–– e.g. the HMIS shouldn't be calling any of the "normal" warehouse controllers–– so it also kind of enforces that separation in the code. The GraphQL controller(s) will be using `current_hmis_api_user`, and everything else uses `current_user`.

Reply:
And, in case I misunderstood your question... `current_user` is `nil` here because the user is authenticated as an HmisApiUser, not a User.